File: backend/app/schemas/generation.py
"""app/schemas/generation.py — API request/response shapes for document generation."""
from __future__ import annotations

import uuid

from pydantic import BaseModel


class GenerateRequest(BaseModel):
    resume_id: uuid.UUID
    job_id: uuid.UUID
    application_id: uuid.UUID | None = None


class EmailContent(BaseModel):
    subject: str
    body: str


class GeneratedDocumentsResponse(BaseModel):
    resume_markdown: str | None = None
    cover_letter: str | None = None
    application_email: EmailContent | None = None
    changes_made: list[str] = []
    truthfulness_check_passed: bool

File: frontend/src/types/generation.ts
export interface EmailContent {
  subject: string;
  body: string;
}

export interface GeneratedDocuments {
  resume_markdown?: string | null;
  cover_letter?: string | null;
  application_email?: EmailContent | null;
  changes_made: string[];
  truthfulness_check_passed: boolean;
}
